devices/At24cxx/At24Base.cs

```diff
@@ -0,0 +1,185 @@
+        /// Helper function to create a buffer with the first two bytes already populated by the device address.
+        /// </summary>
+        /// <param name="address">The device address.</param>
+        /// <param name="length">The length of data the buffer is required to hold.</param>
+        /// <returns>A buffer with the first two bytes populated with the device address.</returns>
+        private static byte[] CreateWriteBuffer(int address, int length)
+        {
+            // Create buffer with an additional two bytes to hold the address.
+            byte[] buffer = new byte[length + 2];
+
+            // Store the address in the first two bytes.
+            buffer[0] = (byte)((address >> 8) & 0xFF);
+            buffer[1] = (byte)(address & 0xFF);
+
+            return buffer;
+        }
+
+        /// <summary>
+        /// Helper function to verify a memory address falls inside the accessible memory range for a At24c device.
+        /// </summary>
+        /// <param name="address">The memory address.</param>
+        /// <param name="availableMemory">The amount of memory available on the device.</param>
+        private static void VerifyMemoryAddress(int address, int availableMemory)
+        {
+            if (address < 0 || address > (availableMemory - 1))
+            {
+                throw new ArgumentOutOfRangeException(nameof(address), "Address must be within addressable memory range.");
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="At24Base" /> class.
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device to use for communication.</param>
+        /// <param name="pageSize">The device page size in bytes.</param>
+        /// <param name="pageCount">The number of pages on the device.</param>
+        public At24Base(I2cDevice i2cDevice, ushort pageSize, ushort pageCount)
+        {
+            if (i2cDevice == null)
+            {
+                throw new ArgumentNullException(nameof(i2cDevice));
+            }
+
+            _i2cDevice = i2cDevice;
+            PageSize = pageSize;
+            PageCount = pageCount;
+            Size = pageSize * pageCount;
+        }
+
+        /// <summary>
+        /// Reads a single byte from the device at the address following the last byte read or written.
+        /// </summary>
+        /// <returns>The byte read from the device.</returns>
+        public byte ReadByte()
+        {
+            return _i2cDevice.ReadByte();
+        }
+
+        /// <summary>
+        /// Reads a single byte from the device at the given address.
+        /// </summary>
+        /// <param name="address">The address to read from.</param>
+        /// <returns>The byte read from the device.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when address falls outside of the accessible range for this device.</exception>
+        public byte ReadByte(int address)
```

Comment:
seems that address is always in the 0-255 range. So why not using a byte rather than an int?

Reply:
Memory address range is chip depended, I guess we could change the address to an unsigned short? I just picked int for the convenience.